pkg/apis/extensions/validation/validation_test.go

```diff
@@ -1929,3 +1929,76 @@ func newBool(val bool) *bool {
 	*p = val
 	return p
 }
+
+func TestValidateStorageClass(t *testing.T) {
+	successCases := []extensions.StorageClass{
+		{
+			// Empty type and parameters
+			ObjectMeta:            api.ObjectMeta{Name: "foo"},
+			ProvisionerType:       "",
+			ProvisionerParameters: map[string]string{},
+		},
+		{
+			// nil parameters
+			ObjectMeta:      api.ObjectMeta{Name: "foo"},
+			ProvisionerType: "",
+		},
+		{
+			// some parameters
+			ObjectMeta:      api.ObjectMeta{Name: "foo"},
+			ProvisionerType: "kubernetes.io/foo-provisioner",
+			ProvisionerParameters: map[string]string{
+				"kubernetes.io/foo-parameter": "free/form/string",
+				"foo-parameter":               "free-form-string",
+				"foo-parameter2":              "{\"embeded\": \"json\", \"with\": {\"structures\":\"inside\"}}",
+			},
+		},
+	}
+
+	// Success cases are expected to pass validation.
+	for k, v := range successCases {
+		if errs := ValidateStorageClass(&v); len(errs) != 0 {
+			t.Errorf("Expected success for %d, got %v", k, errs)
+		}
+	}
+
+	// generate a map longer than maxProvisionerParameterSize
+	longParameters := make(map[string]string)
+	totalSize := 0
+	for totalSize < maxProvisionerParameterSize {
```

Comment:
i couldn't figure out a clever way to transform strings.Repeat() to a map that would simplify the code any.